python/libff/kaas/kaas.py

```diff
@@ -126,9 +125,21 @@ def __init__(self, library, kernel, gridDim, blockDim, sharedSize=0, literals=[]
         self.sharedSize = sharedSize 
 
         self.literals = literals
-        self.inputs = inputs 
-        self.temps = temps 
-        self.outputs = outputs 
+        self.arguments = []
+        self.type_list = []
+        self.inputs = []
+        self.outputs = []
+        self.temps = []
+        for i in range(len(arguments)):
+            arg = arguments[i]
```

Comment:
not really important but I'd just assign arguments to self.arguments instead of doing it one by one here. I usually assign the argument early and then iterate self.arguments. This makes it easier to change stuff in the future (you might need to modify arguments before iterating it someday). Mostly just a style thing though.

Reply:
Despite the naming convention, self.arguments is a little different from the parameter arguments in that the parameter is a list of tuples. So a loop is required to extract the first element of each tuple anyways, and I'd rather do it in this loop than have a separate one. If it ends up being a problem this is easily changed.

Reply:
makes sense.